cinema/views.py

```diff
@@ -97,25 +100,42 @@ def get_queryset(self):
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
         if self.action == "retrieve":
             return MovieDetailSerializer
-
+        if self.action == "upload_image":
+            return MovieImageSerializer
```

Comment:
Code Efficiency: Don't use `serializers.CharField` while adding an image. Use `serializers.ImageField` instead. It seems you're using the correct serializer for image upload, but make sure you're not using `serializers.CharField` elsewhere in your code when dealing with images.